### PR TITLE
[Do not merge] Regenerate Chat SDK

### PR DESCRIPTION
**Pull request by knvsl**
I have rebased the feature branch onto master and then regenerated Chat SDK with the changes from the Autorest refactor.

- Commented relevant files - updated tests and applied a few manual fixes in the generated code
- I will force push to the feature branch after review, see the history here: https://github.com/knvsl/azure-sdk-for-ios/commits/knvsl/regenerate-chat

### CODE REVIEW

File: sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift [hunk 1]
```diff
@@ -16,94 +16,56 @@ import Foundation
...
     ) {
-        // Construct URL
-        let urlTemplate = "/chat/threads/{chatThreadId}/readreceipts"
-        let pathParams = [
-            "$endpoint": client.endpoint.absoluteString,
-            "chatThreadId": chatThreadId
-        ]
-        // Construct query
-        let queryParams: [QueryParameter] = [
-            ("api-version", "2020-09-21-preview2")
```
Comment (knvsl) [inline sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift:69]: Generated code produces apiVersion, changed to api-version

File: sdk/communication/AzureCommunicationChat/Source/Generated/Chat.swift [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+    public func getChatThread(
+        chatThreadId: String,
+        withOptions options: GetChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<ChatThread>
+    ) {
+        // Create request parameters
+        let params = RequestParameters(
+            (.path, "chatThreadId", chatThreadId, .encode),
+            (.uri, "endpoint", client.endpoint.absoluteString, .skipEncoding),
+            (.query, "api-version", "2020-11-01-preview3", .encode),
```
Comment (knvsl) [inline sdk/communication/AzureCommunicationChat/Source/Generated/Chat.swift:312]: Same as with ChatThreadOperation, changed apiVersion to api-version throughout

File: sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift [hunk 3]
```diff
@@ -852,58 +705,47 @@ public final class AzureCommunicationChatService {
...
+            (.query, "api-version", "2020-11-01-preview3", .encode),
+            (.header, "Content-Type", "application/merge-patch+json", .encode),
+            (.header, "Accept", "application/json", .encode)
+        )
 
-        // Construct headers
-        var headers = HTTPHeaders()
-        headers["Content-Type"] = "application/json"
-        headers["Accept"] = "application/json"
         // Construct request
```
Comment (knvsl) [inline sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift:717]: For both update methods removed usage of MergePatch(), also changed self.options.logger to client.options.logger 

https://github.com/knvsl/azure-sdk-for-ios/blob/ef29c1fe9edaa18848d01f4db7044904731bf294/sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift#L731

**Comment by tjprescott** [comment]
Something seems wrong with the rebase, as files that (I'm assuming) were not changed are being shown with a diff.

File: sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift (hunk 3 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationChat/Source/Generated/ChatThreadOperation.swift:717]: These should be opened as issues in autorest, not manually changed in the generated code.

**Review by tjprescott — CHANGES_REQUESTED** [review]
Issues with the generated code need to be opened as issues in autorest.swift.

File: sdk/communication/AzureCommunicationChat/Source/Generated/Chat.swift (hunk 2 above)
Comment (knvsl) [reply sdk/communication/AzureCommunicationChat/Source/Generated/Chat.swift:312]: yup! I am opening an issue now

**Comment by knvsl** [comment]
Autorest.swift issue here: https://github.com/Azure/autorest.swift/issues/294

**Comment by knvsl** [comment]
> Something seems wrong with the rebase, as files that (I'm assuming) were not changed are being shown with a diff.

hmm, I just did rebase onto master and had to resolve several conflicts (no merge commits added). I will not be merging this PR into feature/AzureCommunication but will have to force-push it. Let me know if this is the wrong approach though

The commit history we would end up with is here: https://github.com/knvsl/azure-sdk-for-ios/commits/knvsl/regenerate-chat
Targeting this PR against master shows only Chat files changed as expected

**Review by tjprescott — APPROVED** [review]
Okay to merge this so long as the final PR to master has 100% generated code in the Generated folder.

**Comment by knvsl** [comment]
force pushed to feature/AzureCommunication

**Pull request merged**